components/insight/SRC/org/openmicroscopy/shoola/env/data/views/calls/ThumbnailLoader.java

```diff
@@ -235,24 +218,15 @@ public void doCall() throws Exception {
                         try {
                             store = getThumbnailStore(pxd);
                             handleBatchCall(store, pxd, userId);
-                        } catch (DSOutOfServiceException e) {
-                            // connection issue, rethrow, triggers Insight to reconnect
-                            throw e;
-                        }
-                        catch (Throwable e) {
-                            // Can't get access to thumbnail store
-                            Image thumbnail = Factory
-                                    .createDefaultThumbnail("Can't render\n[Error "+Error.NO_THUMBNAILSTORE.code+"]");
-                            currentThumbnail = new ThumbnailData(pxd.getImage()
-                                    .getId(), thumbnail, userId, true);
+                        } catch (DSAccessException | ServerError e) {
+                            currentThumbnail = getErrorIcon();
```

Comment:
That's the issue for the ClassCastException noticed by @mtbc . Although an 'Object' currentThumbnail is assumed to be a "ThumbnailData'.

Reply:
Well spotted!